the_debuginator.h

```diff
@@ -1552,6 +1581,8 @@ void debuginator_update_filter(TheDebuginator* debuginator, const char* wanted_f
 		fallback->parent->folder.hot_child = fallback;
 	}
 
+	debuginator__set_item_total_height_recursively(debuginator->root, debuginator->item_height);
```

Comment:
I had another look, and correct me if I'm wrong, but isn't that what this does? It calls adjust_num_visible_children on the parents of all elements that get filtered (but it kind of has to do that), then does a final (single) set_item_total_height_recursively on the root. Adjusting the child count isn't very expensive (although it is called a lot), and I don't see a simple way around adjusting the final height on the root?